### PR TITLE
Adds args field to tools

### LINKED ISSUE #2395: [REQUEST] configuration of default/suggested command line arguments for tools
**Is your feature request related to a problem? Please describe.**

I run a local compiler-explorer instance on an HPC cluster and I have a few helpful tools configured in e.g. `etc/config/c.defaults.properties`.
Example: run the binary using a scheduler like SLURM (uses the readelf-tool class)
I would like to provide default command line arguments for the tool but 
```
tool.name.options=...
```
does not seem to have an effect here.
The tool `>_Arguments` (provided by the user in the browser) work fine.

**Describe the solution you'd like**

Use the tool `options` as command line arguments if the user does not specify arguments.
Or fill the arguments input box with the "default" options from the configuration file.

For tools where the user should not be able to change the arguments, the `>_Arguments` field could still show them ->
would make it more transparent what happens in the background...

**Describe alternatives you've considered**

Provide a wrapper shell script for the tool that fills in arguments if none are given - or users just need to know useful arguments.

**Additional context**

I assumed this could be implemented quite easily but couldn't figure out where to set this (I'm not really experienced in node.js etc)
Or this already works but I didn't know how to use it...

### PR DESCRIPTION
**Pull request by RubenRBS**
This args field is show in the UI inside the Arguments input.

I decided to make it a PR because I'm rusty as heck and might have broken something unintentionally :)

Closes #2395

### CODE REVIEW

File: lib/tooling/clang-format-tool.js [hunk 1]
```diff
@@ -37,9 +37,9 @@ export class ClangFormatTool extends BaseTool {
         const options = compilationInfo.options;
         const dir = path.dirname(sourcefile);
 
-        const compileFlags = options.filter(option => (option !== sourcefile));
+        let compileFlags = options.filter(option => (option !== sourcefile));
```
Comment (mattgodbolt) [inline lib/tooling/clang-format-tool.js:40]: Super nitpick: Why this change? Just asking - any time something goes from `const` (albeit javascript's silly `const` notion) to `let` I have to ask :)

**Review by mattgodbolt — APPROVED** [review]
Welcome back :)

File: lib/tooling/clang-format-tool.js (hunk 1 above)
Comment (partouf) [reply lib/tooling/clang-format-tool.js:40]: because it gets reassigned below using concat

Comment (mattgodbolt) [reply lib/tooling/clang-format-tool.js:40]: Ah! Got it; and the options is now an array not a single string. We can use `compileFlags.push(*...this.tool.options)` but that's no clearer. Fine as-is! :)
